src/coreclr/jit/rationalize.cpp

```diff
@@ -358,9 +362,28 @@ void Rationalizer::RewriteHWIntrinsicAsUserCall(GenTree** use, ArrayStack<GenTre
             GenTree* op1 = operands[0];
             GenTree* op2 = operands[1];
 
-            if (op2->IsCnsVec() && comp->IsValidForShuffle(op2->AsVecCon(), simdSize, simdBaseType))
+            // Check if the required intrinsics to emit are available.
+            if (!comp->IsValidForShuffle(op2, simdSize, simdBaseType, nullptr))
             {
-                result = comp->gtNewSimdShuffleNode(retType, op1, op2, simdBaseJitType, simdSize);
+                break;
+            }
+
+            bool isShuffleNative = intrinsicId == NI_Vector128_ShuffleNative;
+#if defined(TARGET_XARCH)
+            isShuffleNative = isShuffleNative || intrinsicId == NI_Vector256_ShuffleNative ||
+                              intrinsicId == NI_Vector512_ShuffleNative;
+#elif defined(TARGET_ARM64)
+            isShuffleNative = isShuffleNative || intrinsicId == NI_Vector64_ShuffleNative;
```

Comment:
Here and everywhere else, could you please adjust the code to follow this coding convention: https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/clr-jit-coding-conventions.md#111-logical-and-arithmetic-expressions ?